fuzz/fuzz_targets/differential-new.rs

```diff
@@ -0,0 +1,85 @@
+#![no_main]
+
+use libfuzzer_sys::arbitrary::{Result, Unstructured};
+use libfuzzer_sys::fuzz_target;
+use wasmtime_fuzzing::generators::{DiffValue, SingleInstModule};
+use wasmtime_fuzzing::oracles::engine::{get_exported_function_signatures, DiffEngine};
+use wasmtime_fuzzing::oracles::{diff_wasmtime, differential, engine};
+
+const NUM_INVOCATIONS: usize = 5;
+
+fuzz_target!(|data: [u8; 1024]| {
+    // errors in `run` have to do with not enough input in `data`, which we
+    // ignore here since it doesn't affect how we'd like to fuzz.
+    drop(run(&data));
+});
+
+fn run(data: &[u8]) -> anyhow::Result<()> {
+    let mut u = Unstructured::new(data);
+
+    // Generate the Wasm module. TODO eventually, this should pick between the
+    // single-instruction and wasm-smith modules, but currently the wasm-smith
+    // module generation will eat up all of the random data, leaving none for
+    // the remaining decisions that follow (e.g., choosing an engine, generating
+    // arguments).
+    let module: &SingleInstModule = u.arbitrary()?;
+    let wasm = module.to_bytes();
+    let features = module.to_features();
+
+    // Choose a right-hand side Wasm engine--this will always be Wasmtime. The
+    // order (execute `lhs` first, then `rhs`) is important because, in some
+    // cases (e.g., OCaml spec interpreter), both sides register signal
+    // handlers; Wasmtime uses these signal handlers for catching various
+    // WebAssembly failures. On certain OSes (e.g. Linux x86_64), the signal
+    // handlers interfere, observable as an uncaught `SIGSEGV`--not even caught
+    // by libFuzzer. By always running Wasmtime second, its signal handlers are
+    // registered most recently and they catch failures appropriately. We create
+    // `rhs` first, however, so we have the option of creating a compatible
+    // Wasmtime engine (e.g., pooling allocator memory differences).
+    let rhs = diff_wasmtime::WasmtimeEngine::arbitrary_with_features(&mut u, &features)
+        .expect("unable to build a Wasmtime `rhs` engine");
+
+    // Choose a left-hand side Wasm engine.
+    let lhs = engine::choose(&mut u, &features, &rhs)
+        .expect("unable to choose an engine to fuzz against");
+
+    // Instantiate each engine and try each exported functions with various
+    // values.
+    let mut lhs_instance = lhs
+        .instantiate(&module.to_bytes())
+        .expect("failed to instantiate `lhs` module");
+    let mut rhs_instance = rhs
+        .instantiate(&module.to_bytes())
+        .expect("failed to instantiate `rhs` module");
+    for (name, signature) in get_exported_function_signatures(&wasm)
+        .expect("failed to extract exported function signatures")
+    {
+        let mut invocations = 0;
+        loop {
+            let arguments: Result<Vec<_>> = signature
+                .params
+                .iter()
+                .map(|&t| DiffValue::arbitrary_of_type(&mut u, t.into()))
+                .collect();
```

Comment:
Running the exported functions multiple times will -- in general, between single-instruction modules and fuel limits -- be a lot cheaper than compiling and instantiating the module itself. Therefore, we should make sure we take as much advantage of the work we already did to get here by at least using each of our known values for each argument, and then whatever the fuzzer gives us that is left over can also be used to generate new inputs, but at least this way we know we are exercising every module with a bunch of different interesting argument values.